x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/VerifierTests.java

```diff
@@ -2689,6 +2689,25 @@ public void testNoMetricInStatsByClause() {
         );
     }
 
+    public void testNoDimensionsInAggsOnlyInByClause() {
+        assertThat(
+            error("TS test | STATS count(host) BY bucket(@timestamp, 1 minute)", tsdb),
+            equalTo(
+                "1:11: cannot use dimension field [host] in a time-series aggregation function [count(host)]. "
+                    + "Dimension fields can only be used for grouping in a BY clause. "
+                    + "To aggregate dimension fields, use the FROM command instead of the TS command."
+            )
+        );
+        assertThat(
+            error("TS test | STATS count(to_string(host)) BY bucket(@timestamp, 1 minute)", tsdb),
```

Comment:
I think this doesn't work - can you test with `STATS count(last_over_time(host))`?